flow/surface.h

```diff
@@ -35,22 +35,13 @@ class Surface {
 
   virtual bool IsValid() = 0;
 
-  virtual std::unique_ptr<SurfaceFrame> AcquireFrame(const SkISize& size) = 0;
+  virtual std::unique_ptr<SurfaceFrame> AcquireFrame(int64_t view_id,
```

Comment:
Could you explain why this has a `view_id` parameter? We will have separate surfaces for each view right?

Reply:
Yeah you are right.

Initially I added the view_id to `SurfaceFrame::Submit`, so that surfaces and surface frames can be "view agnostic" until the last second so that they feel more "pure", but it turns out that we want frame buffers to be tied with a view ID so that buffers can be invalidated more easily. I moved `view_id` earlier to `AcquireFrame`, but on a second thought we should just straight up tie surfaces with views anyway.

Reply:
Although I'm wondering if we can assume that all surfaces for a view are of the same size, so that if we invalidate the frame buffer for one surface of a view, then we can also invalidate all other frame buffers of that view.

Reply:
> I'm wondering if we can assume that all surfaces for a view are of the same size

Are you including the macOS embedder's surfaces here? If so, I don't think so: the engine may have one layer for a button and another layer for the app's background. From my understanding, these would translate into macOS embedder surfaces with different dimensions.